yt_dlp/extractor/facebook.py

```diff
@@ -401,9 +417,9 @@ def _extract_from_url(self, url, video_id):
 
         def extract_metadata(webpage):
             post_data = [self._parse_json(j, video_id, fatal=False) for j in re.findall(
-                r'handleWithCustomApplyEach\(\s*ScheduledApplyEach\s*,\s*(\{.+?\})\s*\);', webpage)]
+                r'data-sjs>({.*?ScheduledServerJS.*?})</script>', webpage)]
```

Comment:
This can cause catastrophic backtracking. Try to see if there's a better way. Can still merge if no easy solution is found, but it is something to be aware of.

Reply:
Well, this is potential replacement:
```
r'data-sjs>([^<]+)</script>', webpage)]
```
Just traverse in the next line will have more items to browse through. Not sure if this would provide better performance since `webpage` data that is collected is up to few MBs in size. Not GBs or something.

Reply:
Haven't tested, but I think a relatively easy way would be `r'data-sjs>({(?:(?!ScheduledServerJS).)*?ScheduledServerJS.*?})</script>'`. A cleaner way would be `r'data-sjs>({(?>.*?ScheduledServerJS).*?})</script>'`, but using atomic groups `(?>…)` would unfortunately require Python 3.11.

And similar for the `%s` below.

P.S. Was there something wrong with my other comment in this thread? It says it was hidden.

Reply:
@Cyp I'm not too much of an expert in regex optimisation, but it looks to be more complicated by testing it in https://regex101.com/

Reply:
> P.S. Was there something wrong with my other comment in this thread? It says it was hidden.

I hid the comment because bb5d84c9d2f1e978c3eddfb5ccbe138036682a36 already does what you suggested (you must not be working from master)

Reply:
I was working with a distro version, with some patch files I semi-auto-applied from 7564 and 7890. Looks like one important chunk of the 7564.patch file silently failed to be applied even though other parts were (I can see it in the patch file, but after applying the patch, that part is ignored somehow). Anyway, thanks, good to know it's fixed in master then.